### PR TITLE
chore: use `ghcr.io/kedacore/tests-hey` image in e2e tests for HTTP load testing

### PR DESCRIPTION
**Pull request by zroubalik**
### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

### CODE REVIEW

File: tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go [hunk 1]
```diff
@@ -426,7 +426,7 @@ spec:
   template:
     spec:
       containers:
-      - image: quay.io/zroubalik/hey
+      - image: ghcr.io/kedacore/tests-hey
```
Comment (JorTurFer) [inline tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go:429]: I have updated this file as part of my PR, revert changes in hashicorp file please

**Review by JorTurFer — APPROVED** [review]

File: tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go (hunk 1 above)
Comment (zroubalik) [reply tests/secret-providers/hashicorp_vault/hashicorp_vault_test.go]: done

**Pull request merged**